client/my-sites/people/edit-team-member-form/edit-user-form.jsx

```diff
@@ -108,28 +112,61 @@ class EditUserForm extends React.Component {
 				? Object.assign( changedSettings, { roles: [ changedSettings.roles ] } )
 				: changedSettings
 		);
+
+		if ( true === changedSettings.isExternalContributor ) {
+			requestExternalContributorsAddition( this.props.siteId, this.state.ID );
+		} else if ( false === changedSettings.isExternalContributor ) {
+			requestExternalContributorsRemoval( this.props.siteId, this.state.ID );
+		}
+
 		this.props.recordGoogleEvent( 'People', 'Clicked Save Changes Button on User Edit' );
 	};
 
 	recordFieldFocus = fieldId => () =>
 		this.props.recordGoogleEvent( 'People', 'Focused on field on User Edit', 'Field', fieldId );
 
-	handleChange = event => this.setState( { [ event.target.name ]: event.target.value } );
+	handleChange = event => {
+		const stateChange = { [ event.target.name ]: event.target.value };
+		if (
+			'roles' === event.target.name &&
+			'administrator' !== event.target.value &&
+			true === this.state.isExternalContributor
+		) {
+			stateChange.isExternalContributor = false;
+		} else if (
+			'roles' === event.target.name &&
+			'administrator' === event.target.value &&
+			true === this.props.isExternalContributor
+		) {
+			// restore external contributor if other role is clicked then changed back
+			stateChange.isExternalContributor = true;
+		}
+		this.setState( stateChange );
```

Comment:
Unfortunately the conditions are as minimal as they can be, and we don't want to set `stateChange.isExternalContributor` unless we want it to change. I will try rewriting them in a clearer way, however.

Reply:
it may be helpful to create some helper bindings/variables to remove some noise and also employ some early-abort logic

```js
handleChange = ( { target: { name, value } } ) => {
	if ( 'roles' !== name ) {
		return this.setState( { [ name ]: value } );
	}

	// name === 'roles'
	const isAdmin = 'administrator' === value;
	const { isExternalContributor: stateExternal } = this.state;
	const { isExternalContributor: propsExternal } = this.props;

	const isExternalContributor = (
		( ! isAdmin && stateExternal ) ? false :
		( isAdmin && propExternal ) ? true :
		stateExternal
	);

	return this.setState( {
		[ name ]: value,
		isExternalContributor,
	} );
}
```

Somewhere in there I think the intentionality of the logic is hiding. Maybe we could elaborate on the comment "we may need to change" 

```js
// if we lost admin status and were an external contributor
// then we must lost our external contributor status as well
…

// if we gained admin status and are also supposed to be external
// then we need to record that status in our copy of the state
…

// otherwise external status isn't affected, keep the old value
```